File: Java/Easy/Accepted/prime-number-of-set-bits-in-binary-representation_7.java
class Solution {
    public int countPrimeSetBits(int left, int right) {
        
        int count = 0;
        for (int i = left; i <= right; i++) {
            count += isPrime(Integer.bitCount(i)) ? 1 : 0;
        }
        return count;
    }

    private boolean isPrime(int n) {
        if (n == 2 || n == 3 || n == 5 || n == 7 || n == 11 || n == 13 || n == 17 || n == 19) {
            return true;
        }
        return false;
    }
}

File: Rust/Easy/Accepted/prime-number-of-set-bits-in-binary-representation_7.rs
impl Solution {
    pub fn count_prime_set_bits(left: i32, right: i32) -> i32 {
        
        let mut count = 0;
        for i in left..=right {
            let mut set_bits = 0;
            let mut n = i;
            while n > 0 {
                set_bits += n & 1;
                n >>= 1;
            }
            count += match set_bits {
                2 | 3 | 5 | 7 | 11 | 13 | 17 | 19 => 1,
                _ => 0,
            };
        }
        count
    }
}